app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -72,8 +72,8 @@ def distribution_repositories(host)
 
         if content_view && lifecycle_environment
           Katello::Repository.in_environment(lifecycle_environment).in_content_views([content_view]).
-              where(:distribution_version => host.os.release,
-                    :distribution_arch => host.architecture.name,
+              where(:distribution_version => host.try(:os).try(:release),
```

Comment:
if os is nil we really shouldn't try to find a repository.  Repositories do exist with nil distributions (when they contain no kickstart tree), so this may end up causing a repo to be used that isn't valid for provisioning.  I think just adding an os condition on line 73 should work:

`if content_view && lifecycle_environment && host.os`